src/solver/conv/conv_hip_implicit_gemm_3d_grouped_wrw_xdlops.cpp

```diff
@@ -433,20 +387,43 @@ void PerformanceConfigHipImplicitGemm3DGroupWrwXdlops::HeuristicInit(
 bool PerformanceConfigHipImplicitGemm3DGroupWrwXdlops::SetNextValue(
     const ProblemDescription& problem)
 {
+#if MIOPEN_USE_COMPOSABLEKERNEL
     if(valid_kernels.empty())
     {
-        HeuristicInit(problem);
+        switch(problem.GetInDataType())
+        {
+        case miopenHalf: Init<ck::half_t>(problem); break;
+        case miopenFloat: Init<float>(problem); break;
+        case miopenInt8: Init<int8_t>(problem); break;
+        case miopenBFloat16: Init<ck::bhalf_t>(problem); break;
+        case miopenInt64:
+        case miopenInt32:
+        case miopenFloat8:
+        case miopenBFloat8:
+        case miopenDouble: break;
+        }
         assert(!valid_kernels.empty());
         return true;
     }
```

Comment:
Should it still call HeuristicInit, but then return false if valid_kernels is empty? 
Or maybe just return false if the valid_kernels are empty?

I think it shouldn't return true if it's empty since it'll keep looping on SetNextValue, but also not sure why HeuristicInit was removed here.

Reply:
Yes. Basically the code section I added here does exactly the same in heuristicInit. And you are right, instead of the assertion check, it needs to return false if the valid_kenel is still empty after the initialization.